src/google/protobuf/port_def.inc

```diff
@@ -413,6 +413,26 @@
 #define PROTOBUF_ATTRIBUTE_REINITIALIZES
 #endif
 
+#ifdef PROTOBUF_NO_RETURN
+#error PROTOBUF_NO_RETURN was previously defined
+#endif
+#ifdef __cpp_has_attribute
+#if __cpp_has_attribute(noreturn)
+#define PROTOBUF_NO_RETURN [[noreturn]]
+#else
+#define PROTOBUF_NO_RETURN
+#endif
+#else
+#if defined(__GNUC__) || defined(__clang__)
+#define PROTOBUF_NO_RETURN __attribute__((noreturn))
+#elif defined(_MSC_VER)
```

Comment:
this would be PROTOBUF_MSC_VER_MIN

Reply:
There does not seem to be any information as to when that became a supported thing. So I would just go with "all of them" and leave it as is.